pkg/volume/empty_dir/empty_dir.go

```diff
@@ -66,27 +66,29 @@ func (plugin *emptyDirPlugin) Name() string {
 	return emptyDirPluginName
 }
 
-func (plugin *emptyDirPlugin) CanSupport(spec *api.Volume) bool {
+func (plugin *emptyDirPlugin) CanSupport(spec *volume.InternalVolume) bool {
 	if plugin.legacyMode {
 		// Legacy mode instances can be cleaned up but not created anew.
 		return false
 	}
 
-	if util.AllPtrFieldsNil(&spec.VolumeSource) {
+	if util.AllPtrFieldsNil(spec.PersistentVolumeSource) ||
```

Comment:
EmptyDir has the same issue with a malformed VS (though by this point validation should have caught it).

I think && would preserve the current behavior of a default EmptyDir, if that's what we want.  The || is definitely wrong.